File: trunk/SoundCapture/SoundCaptureDevice.cs
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.DirectX.DirectSound;

namespace SoundCapture
{
    /// <summary>
    /// Capture device.
    /// </summary>
    public class SoundCaptureDevice
    {
        Guid id;

        string name;

        public bool IsDefault
        {
            get { return id == Guid.Empty; }
        }

        /// <summary>
        /// Name of the device.
        /// </summary>
        public string Name
        {
            get { return name; }
        }

        internal Guid Id
        {
            get { return id; }
        }

        internal SoundCaptureDevice(Guid id, string name)
        {
            this.id = id;
            this.name = name;
        }

        public static SoundCaptureDevice[] GetDevices()
        {
            CaptureDevicesCollection captureDevices = new CaptureDevicesCollection();
            List<SoundCaptureDevice> devices = new List<SoundCaptureDevice>();
            foreach (DeviceInformation captureDevice in captureDevices)
            {
                devices.Add(new SoundCaptureDevice(captureDevice.DriverGuid, captureDevice.Description));
            }
            return devices.ToArray();
        }

        public static SoundCaptureDevice GetDefaultDevice()
        {
            CaptureDevicesCollection captureDevices = new CaptureDevicesCollection();
            SoundCaptureDevice device = null;
            foreach (DeviceInformation captureDevice in captureDevices)
            {
                if(captureDevice.DriverGuid == Guid.Empty)
                {
                    device = new SoundCaptureDevice(captureDevice.DriverGuid, captureDevice.Description);
                    break;
                }
            }
            if (device == null)
                throw new SoundCaptureException("Default capture device is not found");
            return device;
        }
    }
}

File: vb.net/SoundCapture/SoundCaptureDevice.vb
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.DirectX.DirectSound

''' <summary>
''' Capture device.
''' </summary>
Public Class SoundCaptureDevice
	Private m_id As Guid

	Private m_name As String

	Public ReadOnly Property IsDefault() As Boolean
		Get
			Return m_id = Guid.Empty
		End Get
	End Property

	''' <summary>
	''' Name of the device.
	''' </summary>
	Public ReadOnly Property Name() As String
		Get
			Return m_name
		End Get
	End Property

	Friend ReadOnly Property Id() As Guid
		Get
			Return m_id
		End Get
	End Property

	Friend Sub New(id As Guid, name As String)
		Me.m_id = id
		Me.m_name = name
	End Sub

	Public Shared Function GetDevices() As SoundCaptureDevice()
		Dim captureDevices As New CaptureDevicesCollection()
		Dim devices As New List(Of SoundCaptureDevice)()
		For Each captureDevice As DeviceInformation In captureDevices
			devices.Add(New SoundCaptureDevice(captureDevice.DriverGuid, captureDevice.Description))
		Next
		Return devices.ToArray()
	End Function

	Public Shared Function GetDefaultDevice() As SoundCaptureDevice
		Dim captureDevices As New CaptureDevicesCollection()
		Dim device As SoundCaptureDevice = Nothing
		For Each captureDevice As DeviceInformation In captureDevices
			If captureDevice.DriverGuid = Guid.Empty Then
				device = New SoundCaptureDevice(captureDevice.DriverGuid, captureDevice.Description)
				Exit For
			End If
		Next
		If device Is Nothing Then
			Throw New SoundCaptureException("Default capture device is not found")
		End If
		Return device
	End Function
End Class
